runtime/autoload/man.vim

```diff
@@ -71,16 +65,56 @@ function! man#read_page(ref) abort
   call s:read_page(path)
 endfunction
 
+" Handler for s:system() function.
+function! s:system_handler(jobid, data, event) abort
+  if a:event == 'stdout'
+    let self.output .= join(a:data, "\n")
+  elseif a:event == 'exit'
+    let s:shell_error = a:data
+  endif
+endfunction
+
+" Run a system command and timeout after 30 seconds.
+function! s:system(cmd, ...) abort
+  let stdin = a:0 ? a:1 : ''
+  let opts = {
+        \ 'output': '',
+        \ 'on_stdout': function('s:system_handler'),
+        \ 'on_exit': function('s:system_handler'),
+        \ }
+  let jobid = jobstart(a:cmd, opts)
+
+  if jobid < 1
+    throw printf('Command error %d: %s', jobid, 
+        \ type(a:cmd) == type([]) ? join(a:cmd) : a:cmd)
+  endif
+
+  if !empty(stdin)
+    call jobsend(jobid, stdin)
+  endif
+
+  let res = jobwait([jobid], 30000)
+  if res[0] == -1
+    call jobstop(jobid)
+    throw printf('Command timed out: %s',
+        \ type(a:cmd) == type([]) ? join(a:cmd) : a:cmd)
```

Comment:
No need to handle this either (here, below and above) because we are only going to send a list as the command.